x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/inference/TrainedModelConfigTests.java

```diff
@@ -307,53 +309,61 @@ public void testSerializationWithCompressedLazyDefinition() throws IOException {
-        when(licenseState.isActive()).thenReturn(false);
-        when(licenseState.getOperationMode()).thenReturn(License.OperationMode.BASIC);
 
+        // Reject everything
+        when(licenseState.isAllowedByLicense(any(License.OperationMode.class), anyBoolean(), anyBoolean())).thenAnswer(
+            invocationOnMock -> {
+                final Object[] arguments = invocationOnMock.getArguments();
+                assertTrue((boolean) arguments[1]); // ensure the call is made to require active license
+                return false;
+            }
+        );
         assertFalse(builder.setLicenseLevel(License.OperationMode.ENTERPRISE.description()).build().isAvailableWithLicense(licenseState));
         assertFalse(builder.setLicenseLevel(License.OperationMode.PLATINUM.description()).build().isAvailableWithLicense(licenseState));
+        assertFalse(builder.setLicenseLevel(License.OperationMode.GOLD.description()).build().isAvailableWithLicense(licenseState));
+        // Basic license always works not matter what
         assertTrue(builder.setLicenseLevel(License.OperationMode.BASIC.description()).build().isAvailableWithLicense(licenseState));
+    }
 
-        when(licenseState.isActive()).thenReturn(true);
-        when(licenseState.getOperationMode()).thenReturn(License.OperationMode.ENTERPRISE);
-        assertTrue(builder.setLicenseLevel(License.OperationMode.ENTERPRISE.description()).build().isAvailableWithLicense(licenseState));
-        assertTrue(builder.setLicenseLevel(License.OperationMode.PLATINUM.description()).build().isAvailableWithLicense(licenseState));
-        assertTrue(builder.setLicenseLevel(License.OperationMode.BASIC.description()).build().isAvailableWithLicense(licenseState));
-        assertTrue(builder.setLicenseLevel(License.OperationMode.GOLD.description()).build().isAvailableWithLicense(licenseState));
-
-        when(licenseState.isActive()).thenReturn(false);
-        assertFalse(builder.setLicenseLevel(License.OperationMode.ENTERPRISE.description()).build().isAvailableWithLicense(licenseState));
-        assertFalse(builder.setLicenseLevel(License.OperationMode.PLATINUM.description()).build().isAvailableWithLicense(licenseState));
-        assertTrue(builder.setLicenseLevel(License.OperationMode.BASIC.description()).build().isAvailableWithLicense(licenseState));
-        assertFalse(builder.setLicenseLevel(License.OperationMode.GOLD.description()).build().isAvailableWithLicense(licenseState));
+    public void testActivePlatinumLicenseAlwaysWorks() {
+        TrainedModelConfig.Builder builder = createTestInstance(randomAlphaOfLength(10));
+        XPackLicenseState licenseState = mock(XPackLicenseState.class);
 
-        when(licenseState.isActive()).thenReturn(true);
-        when(licenseState.getOperationMode()).thenReturn(License.OperationMode.PLATINUM);
+        // Active Platinum license is considered as highest and should always work
```

Comment:
This isn't true any more, right?
I mean, technically right now there's no functional difference between Platinum and Enterprise modes, but `Platinum` isn't actually the highest now.

Reply:
You are right. I was trying to hint the same thing in the comment, i.e. "considered as ...". But it is too subtle to understand. I'll reword it.

Reply:
[Updated](https://github.com/elastic/elasticsearch/pull/52118/files#diff-ffdd5f9c51b0efbbeb3226c81d69bb20R245)